src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -3593,7 +3593,7 @@ else if (option.isInLocalDCOnly())
             }
             else
             {
-                Iterables.addAll(option.getRanges(), getLocalReplicas(keyspace).asRanges());
+                Iterables.addAll(option.getRanges(), getLocalReplicas(keyspace).filter(Replica::isFull).asRanges());
```

Comment:
bq. So this just means when you don't specify -pr we only repair the full ranges?

Correct

bq. I'm not actually clear on what operational procedures are for running repair on a cluster. Is it rolling invocations of repair with -pr? What is the use case for repair without -pr, to fully repair the node with the data it needs if you think it isn't consistent?

You can go either way. I think it's easiest to reason about rolling -pr repairs, and some people probably do. Even if you run -pr repairs, you may want to repair all ranges associated with a node in some cases. First one that comes to mind would be if you deleted a corrupt sstable and didn't want to completely replace a node.

bq. In that scenario it seems like it won't be a valid transient replica for that range until the next incremental repair completes for its transient ranges. That makes this a slight change in what it does.

I'm not really clear on what you're saying here. How does a node become an invalid transient replica, and what is there a slight change in?

Reply:
In the old world if you ran repair without -pr it repaired all ranges on that node and you could no the node was consistent for any range it was responsible.

With this change were we don't repair ranges we transiently replicate this is no longer true.

It seems like not coordinating repairs of transient replicas may not be the right thing to do unlress -pr is specified?

Reply:
You could probably come up with equally plausible arguments for or against coordinating repairs for locally transient ranges. On one hand, they’re replicated by the coordinating node. On the other, they're inherently inconsistent, and repairing won’t make the local transient range consistent with anything else.

From a technical perspective, coordinating repairs for transient ranges is riskier. The local node has this sort of implicit status in repairs and is actually not included in any (well, most) lists of participating endpoints. Because of this, there are a lot of places where we make assumptions about the role of the coordinating node. If we coordinate repairs for transient ranges, the places where we make assumptions about the coordinating node need to be identified and possibly adjusted.